prov/gni/src/gnix_mr_cache.c

```diff
@@ -373,6 +622,15 @@ static inline int __insert_entry_into_stale(
 	RbtStatus rc;
 	int ret = 0;
 
+	if ((entry->state == GNIX_CES_INUSE_UNMAPPED) ||
+	    (entry->state == GNIX_CES_STALE_UNMAPPED)) {
+		GNIX_INFO(FI_LOG_MR, "entry (%p) unmapped, not inserting"
+			  " into stale %llx:%llx", entry,
```

Comment:
I have a note up top that says we should reevaluate how we maintain states and flags, possibly combining all in a single state variable.  Let's talk more about it later.